flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/enumerator/PulsarSourceEnumerator.java

```diff
@@ -96,12 +98,18 @@ public PulsarSourceEnumerator(
         this.sourceConfiguration = sourceConfiguration;
         this.context = context;
         this.splitAssigner = createAssigner(stopCursor, sourceConfiguration, context, enumState);
+        this.metricGroup = context.metricGroup();
     }
 
     @Override
     public void start() {
         rangeGenerator.open(sourceConfiguration);
 
+        // Expose the split assignment metrics if Flink has supported.
+        if (metricGroup != null) {
+            metricGroup.setUnassignedSplitsGauge(splitAssigner::getUnassignedSplitCount);
```

Comment:
It seems this function call is always noop from the current codebase.

Reply:
Yep. Flink may add this support one day. I just add it for making sure all the metrics have been exposed.

Reply:
OR should I just remove it now?

Reply:
Nope. I think it's OK with this intention.